### DISCUSSION TITLE
Interested in adding `pluck`?

### DISCUSSION THREAD

**Discussion by crishoj** [radashi-org/radashi#281] [Ideas]
I frequently find myself needing a `pluck` function to extract a property from an object array:

```ts
export function pluck<T extends object, K extends keyof T>(
  objects: T[],
  key: K,
): T[K][] {
  return objects.map(obj => obj[key])
}
```

Example use:

```ts
    const people = [
      { name: 'John', age: 20 },
      { name: 'Jane', age: 21 },
      { name: 'Joe', age: 22 },
      { name: 'Jill', age: 23 },
      { name: 'Jim', age: 24 },
    ]
    _.pluck(people, 'name')
```

It was present in an earlier Lodash version, but since removed: https://lodash.com/docs/3.10.1#pluck. 

I have a PR ready, but thought I would gauge the interest here first before submitting.

**Comment by aleclarson** [comment]
Hey! Thanks for the suggestion.

This breaks our rule of “avoid syntax sugar”, since the vanilla JavaScript equivalent is simple and straight-forward.

What do you think?

**Reply by crishoj** [reply]
I concur that the vanilla JS equivalent with `.map` is just fine.
